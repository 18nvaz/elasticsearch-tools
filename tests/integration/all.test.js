const elasticsearch = require('elasticsearch');
const { execSync } = require('child_process');
const insertData = require('./insert-data');

const host = 'http://localhost:20202';
const client = new elasticsearch.Client({ host });
const commands = ['aliases', 'bulk', 'mappings', 'settings'];
const libdir = `../../lib/`;
const workingdir = `../fixtures/working`;
const importdir = `../fixtures/import-tests/`;

const run = (type, cmd) => `node ${libdir}/es-${type}-${cmd} --url ${host}`;

const testAllExports = async (prefix) => {
  commands.forEach(command => {
    const cmd = `${run('export', command)} --file ${workingdir}/${prefix}-${command}.txt`;
    try {
      const result = execSync(cmd, { cwd: __dirname }).toString();
      expect(result).toMatchSnapshot();
    } catch (err) {
      expect(err.message).toMatchSnapshot();
    }
  });
};

beforeAll(async () => {
  // eslint-disable-next-line
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 300000;
  await client.indices.delete({ index: '*' });
});

describe('with empty db', async () => {
  test('client.indices.stats', async () => {
    const stats = await client.indices.stats();
    expect(stats.indices).toEqual({});
  });
  test('test all exports', async () => {
    await testAllExports('1-empty');
  });
});

describe('insert data with node lib', async () => {
  test('insert data', async () => {
    await insertData(client, 'test-lib', 'groups');
  });
  test('client.indices.stats', async () => {
    const stats = await client.indices.stats();
    expect(Object.keys(stats.indices)).toMatchSnapshot();
  });
  test('test all exports', async () => {
    await testAllExports('2-somedata');
  });
});

describe('delete all indices again', async () => {
  test('delete indices', async () => {
    await client.indices.delete({ index: '*' });
  });
  test('client.indices.stats', async () => {
    const stats = await client.indices.stats();
    expect(stats.indices).toEqual({});
  });
  test('test all exports', async () => {
    await testAllExports('3-emptyagain');
  });
});

describe('importing from files', async () => {
  test('test all imports', async () => {
    ['settings', 'aliases', 'mappings', 'bulk'].forEach(command => {
      const result = execSync(`${run('import', command)} --file ${importdir}/${command}.txt`, { cwd: __dirname }).toString();
      expect(result).toMatchSnapshot();
    });
  });
  test('client.indices.stats', async () => {
    const stats = await client.indices.stats();
    expect(Object.keys(stats.indices)).toMatchSnapshot();
  });
  test('test all exports', async () => {
    await testAllExports('4-afterimports');
  });
});
