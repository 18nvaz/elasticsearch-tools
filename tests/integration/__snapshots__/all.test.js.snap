// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete all indices again test all exports 1`] = `
"Exporting aliases for:  []
Done!
"
`;

exports[`delete all indices again test all exports 2`] = `
"Command failed: node ../../lib//es-export-bulk --url http://localhost:20202 --file ../fixtures/working/3-emptyagain-bulk.txt

  error: \\"no results were returned, so exiting.\\"
"
`;

exports[`delete all indices again test all exports 3`] = `
"Exporting mappings for:  []
Done!
"
`;

exports[`delete all indices again test all exports 4`] = `
"Exporting settings for:  []
Done!
"
`;

exports[`importing from files client.indices.stats 1`] = `
Array [
  "test-lib",
]
`;

exports[`importing from files test all exports 1`] = `
"Exporting aliases for:  [ 'test-lib' ]
Done!
"
`;

exports[`importing from files test all exports 2`] = `
"Done!
"
`;

exports[`importing from files test all exports 3`] = `
"Exporting mappings for:  [ 'test-lib' ]
Done!
"
`;

exports[`importing from files test all exports 4`] = `
"Exporting settings for:  [ 'test-lib' ]
Done!
"
`;

exports[`importing from files test all imports 1`] = `
"Creating Indexes (if needed):
  [index=\\"test-lib\\"] - creating
Creating settings:
  [index=\\"test-lib\\", prop=\\"index\\"]
Done!
"
`;

exports[`importing from files test all imports 2`] = `
"Creating Indexes (if needed):
Creating aliases:
Done!
"
`;

exports[`importing from files test all imports 3`] = `
"Creating Indexes (if needed):
  [index=\\"test-lib\\"] - exists
Creating mappings:
  [index=\\"test-lib\\", prop=\\"groups\\"]
Done!
"
`;

exports[`importing from files test all imports 4`] = `
"Pre-Processing file of size: 21.5 KB
Starting bulk imports with batches of 20000 lines.
Complete!
"
`;

exports[`insert data with node lib client.indices.stats 1`] = `
Array [
  "test-lib",
]
`;

exports[`insert data with node lib test all exports 1`] = `
"Exporting aliases for:  [ 'test-lib' ]
Done!
"
`;

exports[`insert data with node lib test all exports 2`] = `
"Done!
"
`;

exports[`insert data with node lib test all exports 3`] = `
"Exporting mappings for:  [ 'test-lib' ]
Done!
"
`;

exports[`insert data with node lib test all exports 4`] = `
"Exporting settings for:  [ 'test-lib' ]
Done!
"
`;

exports[`with empty db test all exports 1`] = `
"Exporting aliases for:  []
Done!
"
`;

exports[`with empty db test all exports 2`] = `
"Command failed: node ../../lib//es-export-bulk --url http://localhost:20202 --file ../fixtures/working/1-empty-bulk.txt

  error: \\"no results were returned, so exiting.\\"
"
`;

exports[`with empty db test all exports 3`] = `
"Exporting mappings for:  []
Done!
"
`;

exports[`with empty db test all exports 4`] = `
"Exporting settings for:  []
Done!
"
`;
